compiler/CMakeLists.txt

```diff
@@ -129,7 +129,8 @@ install (FILES ${output}
     DESTINATION bin)
 
 if(EXISTS "${completion_dir}" AND IS_DIRECTORY "${completion_dir}")
-    install (FILES ${completion}
-        RENAME gbc
-        DESTINATION ${completion_dir})
+    install(CODE "execute_process(COMMAND bash -c \"[[ ! -w ${completion_dir} ]] ||
```

Comment:
What happens if the install destination is "etc/bash_completion.d/" instead of "/etc/bash_completion.d/"? Is CMake smart enough to append the prefix in that case? (I expect the IS_DIRECTORY check will need to go away.) If so, that solution seems more inline with CMake's way of thinking and doesn't rely on bash being installed.

Reply:
The destination is hardcoded in this CMakeLists.txt to "/etc/bash_completion.d" (line 107). There isn't any cmake magic in changing the path to etc. This is why a non-root install fails.

`set (completion_dir /etc/bash_completion.d)`

Reply:
What happens if you change that line to the following?

```
set (completion_dir etc/bash_completion.d)
```

If the initial slash is removed, CMake should start installing relative to the prefix.

From the CMake documentation for [install](https://cmake.org/cmake/help/v3.0/command/install.html):

> Specify the directory on disk to which a file will be installed. If a full path (with a leading slash or drive letter) is given it is used directly. If a relative path is given it is interpreted relative to the value of the CMAKE_INSTALL_PREFIX variable. The prefix can be relocated at install time using the DESTDIR mechanism explained in the CMAKE_INSTALL_PREFIX variable documentation.

Reply:
Yes, I agree it will install relative to CMAKE_INSTALL_PREFIX. I limited myself to fix the very narrow issue of non-root installs to the absolute path of "/etc/bash_completion.d". Changing the path to be relative will change the behavior of the install and that was not my goal. When no CMAKE_INSTALL_PREFIX is set, it would install to "/usr/local/etc/bash_completion.d". If you and your team is fine with that, I can send a PR for that.

Reply:
Just to be thorough, I confirmed that it installs to the relative path of CMAKE_INSTALL_PREFIX if you remove the first slash. Please let me know your preference.

Reply:
My preference would be to install it under the prefix. If the administrator of the system wants it under /etc, they can symlink it after installing.